### ISSUE TITLE
MalformedTreeException (Columba-CCFinder)

### ISSUE THREAD

**Issue by dmazinanian** [tsantalis/JDeodorant#37]
The exception is thrown, but the analysis continues.

Clone # 315 (Group 3242, Pair 1-2): org.columba.mail.gui.message.command.OpenAttachmentCommand#execute (row 819) and org.columba.mail.gui.message.command.SaveAttachmentCommand#execute (row 820)

```
org.eclipse.text.edits.MalformedTreeException: Overlapping text edits
10:56:56.407 [main] INFO  c.c.j.e.commandline.Application - Started refactoring
10:56:56.407 [main] INFO  c.c.j.e.commandline.Application - Finished Refactoring
    at org.eclipse.text.edits.TextEdit$InsertionComparator.compare(TextEdit.java:119)
    at org.eclipse.text.edits.TextEdit$InsertionComparator.compare(TextEdit.java:1)
    at java.util.Collections.indexedBinarySearch(Collections.java:332)
    at java.util.Collections.binarySearch(Collections.java:320)
    at org.eclipse.text.edits.TextEdit.computeInsertionIndex(TextEdit.java:794)
    at org.eclipse.text.edits.TextEdit.internalAdd(TextEdit.java:779)
    at org.eclipse.text.edits.TextEdit.addChild(TextEdit.java:333)
    at gr.uom.java.jdeodorant.refactoring.manipulators.ExtractCloneRefactoring.finalizeCloneExtraction(ExtractCloneRefactoring.java:2348)
    at gr.uom.java.jdeodorant.refactoring.manipulators.ExtractCloneRefactoring.apply(ExtractCloneRefactoring.java:348)
    at gr.uom.java.jdeodorant.refactoring.manipulators.ExtractCloneRefactoring.checkFinalConditions(ExtractCloneRefactoring.java:4858)
    at ca.concordia.jdeodorant.eclipse.commandline.Application.testRefactoring(Application.java:669)
    at ca.concordia.jdeodorant.eclipse.commandline.Application.start(Application.java:185)
    at org.eclipse.equinox.internal.app.EclipseAppHandle.run(EclipseAppHandle.java:196)
    at org.eclipse.core.runtime.internal.adaptor.EclipseAppLauncher.runApplication(EclipseAppLauncher.java:134)
    at org.eclipse.core.runtime.internal.adaptor.EclipseAppLauncher.start(EclipseAppLauncher.java:104)
    at org.eclipse.core.runtime.adaptor.EclipseStarter.run(EclipseStarter.java:388)
    at org.eclipse.core.runtime.adaptor.EclipseStarter.run(EclipseStarter.java:243)
    at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
    at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
    at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
    at java.lang.reflect.Method.invoke(Method.java:498)
    at org.eclipse.equinox.launcher.Main.invokeFramework(Main.java:673)
    at org.eclipse.equinox.launcher.Main.basicRun(Main.java:610)
    at org.eclipse.equinox.launcher.Main.run(Main.java:1519)
    at org.eclipse.equinox.launcher.Main.main(Main.java:1492)
```

**Comment by tsantalis** [comment]
Partially fixed in commit [16a063333511c18a76ef038ba0f3c6064e77b5c7](https://github.com/tsantalis/JDeodorant/commit/16a063333511c18a76ef038ba0f3c6064e77b5c7)

Overlapping text edits is fixed, but there are still compilation errors after the refactoring
1. One of the classes containing the clones is also the common superclass, in which the clones should be extracted. In that case, any common field to be pulled up already exists in the common superclass, and only its access modifier should be updated to protected (if it is private).
2. One of the parameters that the extracted method should have corresponds to a local variable in one of the clones, and a field in the other clone. For some reason the parameter is not introduced.

**Comment by tsantalis** [comment]
Point 1 fixed in commit [bed9cd6f554aa22a1c8ab64bdc9175cf05ad9ce3](https://github.com/tsantalis/JDeodorant/commit/bed9cd6f554aa22a1c8ab64bdc9175cf05ad9ce3)

**Comment by tsantalis** [comment]
Point 2 fixed in commit [ea52d96b641a455e7ab2a8df253ca959f9b0458d](https://github.com/tsantalis/JDeodorant/commit/ea52d96b641a455e7ab2a8df253ca959f9b0458d)
